indico/modules/events/registration/logging.py

```diff
@@ -0,0 +1,56 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2020 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+from __future__ import unicode_literals
+
+from flask import session
+
+from indico.core import signals
+from indico.modules.events import EventLogKind, EventLogRealm
+from indico.modules.events.registration.models.registrations import RegistrationState
+
+
+def connect_log_signals():
+    signals.event.registration_checkin_updated.connect(log_registration_check_in)
+    signals.event.registration_state_updated.connect(log_registration_updated)
+
+
+def log_registration_check_in(sender, *args, **kwargs):
+    """Log a registration check-in action to the event log."""
+    registration = sender
```

Comment:
just call the arg `registration`; sender is a positional arg and not a kwargs. there's also never going to be any other posarg, so `*args` is not needed

Reply:
fixed

Reply:
BTW, for consistency i would like to keep that all the signal handler with the same signature, but, since this will be a "core"code, it's up to you.

Reply:
I think for the `sender` it makes sense to use the name that suits the object, and not the meaningless `sender` name. `*args` would just help shadowing problems in case there are ever new posargs (which shouldn't happen, unlike new kwargs!)

Reply:
Kept `**kwargs` 👌